### PR TITLE
docs(ApplicationLauncher): add examples for disabled, align right, an…

### LINKED ISSUE #2576: AppLauncher: Add support for right and top alignment
This PR was merged in Core: https://github.com/patternfly/patternfly-next/pull/2081

### PR DESCRIPTION
**Pull request by evwilkin**
…d top

**What**:

Fixes #2576 

**Additional issues**:

### CODE REVIEW

File: packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.md [hunk 1]
```diff
@@ -62,7 +62,169 @@ class SimpleApplicationLauncher extends React.Component {
...
+    this.state = {
+      isOpen: false
+    };
+    this.onToggle = isOpen => {
+      this.setState({
+        isOpen
+      });
+    };
+    this.onSelect = event => {
+      this.setState({
```
Comment (priley86) [inline packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.md]: I think a better practice is to use the function form of `setState` for simple toggles like this:

```
this.setState( prevState => ({ isOpen: !prevState.isOpen }) );
```

https://medium.com/@wisecobbler/using-a-function-in-setstate-instead-of-an-object-1f5cfd6e55d1

Comment (redallen) [reply packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.md]: Quote from the article:

> I am relying on the value of this.state.showForm to modify the next state of my form. In this simple example, we probably would not run into any issues by relying on this value, but you might imagine that as an application gets more complex and there are multiple setState calls happening and queuing up data to be rendered to the DOM, there is a possibility that the actual value of this.state.showForm might not be what you think.

I think it's fine to use `setState({})` for simple examples.

File: packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.md [hunk 2]
```diff
@@ -62,7 +62,169 @@ class SimpleApplicationLauncher extends React.Component {
...
+        Unavailable Application
+      </ApplicationLauncherItem>
+    ];
+    return (
+      <ApplicationLauncher
+        onSelect={this.onSelect}
+        onToggle={this.onToggle}
+        isOpen={isOpen}
+        items={appLauncherItems}
+        isDisabled="true"
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.md]:
```suggestion
        isDisabled
```

File: packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.md [hunk 3]
```diff
@@ -62,7 +62,169 @@ class SimpleApplicationLauncher extends React.Component {
...
+        Application 2 (div with onClick)
+      </ApplicationLauncherItem>,
+      <ApplicationLauncherItem key="application_3a" onClick={() => alert('Clicked item 3')}>
+        Application 3 (div with onClick)
+      </ApplicationLauncherItem>,
+      <ApplicationLauncherItem key="disabled_application_4a" isDisabled>
+        Unavailable Application
+      </ApplicationLauncherItem>
+    ];
+    // const style = { marginTop: '100px' };
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.md]: Should be cleaned up here and line 220

File: packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.md (hunk 1 above)
Comment (evwilkin) [reply packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.md]: Leaving as-is for now to match existing examples, we may want to refactor all examples if this change does end up being necessary.

File: packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.md (hunk 2 above)
Comment (evwilkin) [reply packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.md]: @jschuler updated

File: packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.md (hunk 3 above)
Comment (evwilkin) [reply packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.md]: Removed unnecessary code

**Pull request merged**